app/presenters/order_detail_notice_presenter.rb

```diff
@@ -0,0 +1,85 @@
+class OrderDetailNoticePresenter < DelegateClass(OrderDetail)
+
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::OutputSafetyHelper
+
+  def statuses
+    [].tap do |statuses|
+      statuses << Notice.new(:in_review) if in_review?
+      statuses << Notice.new(:in_dispute) if in_dispute?
+      statuses << Notice.new(:can_reconcile) if can_reconcile_journaled?
+      statuses << Notice.new(:in_open_journal) if in_open_journal?
+    end
+  end
+
+  def warnings
+    [].tap do |warnings|
+      warnings << Notice.new(problem_description_key, :warning) if problem?
```

Comment:
Yeah, there's only ever going to be one here. The reason I did it this way was to keep the shape of the `warnings` and `notices` the same. I think it's easier to see they're doing similar logic when they're parallel.